congest integration

toasted226 opened the pull request:
Integrated Congest into Source. There is now a Job Scheduler that can be used from within Source itself and by admins. Scheduled jobs are persisted in sqlite and are recoverable at server startup or by admins. The sqlite tables store the job's status, metadata, args and handler identifiers.

In addition, feeds routes have been added to view all feeds, and to allow a creator to add a new feed. Adding a new feed creates a feed record, extracts RSS feed data and creates an incoming post, then schedules a job to update both the feed and the post every 24 hours.

toasted226 committed:
integrated congest
added jobs and job-metadata tables
added services for sql tables
added handlers for congest

toasted226 committed:
added middleware to include congest job service

toasted226 committed:
updated migrations to include new tables

toasted226 committed:
added and updated db services

toasted226 committed:
added schemas to provider routes

toasted226 committed:
updated reitit with new routes
updated routes to fit naming conventions
added feeds routes

toasted226 committed:
updated parameters for clearer usage

toasted226 committed:
Merge branch 'feat/xml-admin-api' of https://github.com/modulr-software/source-be into feat/congest-integration

toasted226 committed:
hotfixes

merveillevaneck commented on src/source/jobs/oplog.clj:
```diff
@@ -0,0 +1,65 @@
+                  :created-at (:created-at job-metadata)
+                  :sleep (b->i (:sleep job-metadata))}
+        job (services/job ds {:id id})
+        metadata-id (when-not (some? job)
```
these all need to be given docstrings that explainds the scope of each function

merveillevaneck commented on src/source/jobs/oplog.clj:
```diff
@@ -0,0 +1,65 @@
+    (= action "run")
+    (do
+      (update-job! metadata "running")
+      (update-job-metadata! metadata))))
```
please add a docstring to show what this function is meant for

merveillevaneck commented on src/source/middleware/core.clj:
```diff
@@ -22,10 +23,33 @@
   (-> app
       (wrap-store store)))
 
+(defn apply-js [app js]
```
DOCSTRING, js is very ambiguous in the context of web

merveillevaneck reviewed (approved):
please update the persistance layer to only do what it needs to do. instead of makign extensions for the methods in the congest service. we need at most maybe 3 functions, and only 1 of them really needs to be called by the top level, which should be wherever the server is started (a.k.a. a component)

toasted226 committed:
refactored congest integration
removed unnecessary usage of persistance layer services outside oplogger
relying on congest functions instead of wrappers

toasted226 committed:
updated feeds job id

toasted226 committed:
fixed get job route

toasted226 committed:
fixed job routes

toasted226 committed:
updated components to be initialised on server startup and refactored job recovery

merveillevaneck commented on src/source/server.clj:
```diff
@@ -1,26 +1,52 @@
+(defonce ^:private *components (atom nil))
+
+(defn initialise-components! []
+  (if (some? @*components)
```
replace this with a `some` fn invocation where the predicate is true if the component is marked as truthy. this way, you can call a function to determine if the components are on or off. you can then also add dependencies to any component, which should result in the component being "off" if all its dependencies or some of its depencies are "off".

toasted226 committed:
updated server to use a component system

merveillevaneck commented on src/source/server.clj:
```diff
@@ -1,38 +1,103 @@
+  (when (some? (:js @*components))
+    (congest/kill! (:js @*components)))
+  (when (some? (:server @*components))
+    (let [server (:server @*components)]
```
call this server symbol stop-server! instead

merveillevaneck reviewed (approved):
make a cleanup ticket for the PR review discussions and get this merged in in the meantime. good work!

toasted226 committed:
addressed comment
